### PR TITLE
- fix for Swift 5.8 (and swift-parsing 0.12)

### LINKED ISSUE #3: swift-parsing 0.12.0 (Swift 5.8 Support) has breaking change for <= 5.7
Hello, 

It appears that pointfree has release a [newer version](https://github.com/pointfreeco/swift-parsing/releases) of `swift-parsing` which has breaking changes which affect this project. The short term fix might be to lock the package version of `swift-parsing` to `0.11.0` as it automatically pulls the breaking change currently. I'd be happy to look at adding support for Swift 5.8 as well.

Thanks!

### PR DESCRIPTION
**Pull request by ratranqu**
fixes #3

### CODE REVIEW

File: Sources/XMLParser/XMLParser.swift [hunk 1]
```diff
@@ -58,6 +58,7 @@ let commentParser = ParsePrint {
 }
 
 let textParser = ParsePrint {
+    "".utf8 // noop. required to force Whitespace below to take the correct buildExpression
```
Comment (JaapWijnen) [inline Sources/XMLParser/XMLParser.swift]: Can't you add a generic to the Whitespace itself? Or is that not what this is fixing? Or perhaps at the ParsePrint level maybe?

File: Sources/XMLParser/Conversions.swift [hunk 2]
```diff
@@ -34,18 +34,18 @@ extension Conversions {
...
-    struct UnpackXMLElementTuple: Conversion {
+    struct UnpackXMLElement: Conversion {
         @inlinable
         init() { }
-        
+
         @inlinable
-        func apply(_ input: ((String, OrderedDictionary<String, String>), [XML.Node], String)) -> (String, OrderedDictionary<String, String>, [XML.Node]) {
-            (input.0.0, input.0.1, input.1)
+        func apply(_ input: (String, OrderedDictionary<String, String>, [XML.Node], String)) -> (String, OrderedDictionary<String, String>, [XML.Node]) {
```
Comment (JaapWijnen) [inline Sources/XMLParser/Conversions.swift:42]: Can you elaborate why this change of signature is needed?

Comment (JaapWijnen) [reply Sources/XMLParser/Conversions.swift:42]: (Why it's now possible to do without the previous nested tuples)

**Review by JaapWijnen — COMMENTED** [review]
Hey! Thanks for fixing this. I'm surprised this is all that's needed to fix it. Asked for a few clarifications. Other than that looks good. Still have to run the tests on my machine since I don't have CI setup on this repo.

File: Sources/XMLParser/XMLParser.swift (hunk 1 above)
Comment (ratranqu) [reply Sources/XMLParser/XMLParser.swift]: You're right, I added the generic specialisation instead of the extra line.

Comment (ratranqu) [reply Sources/XMLParser/XMLParser.swift]: actually used the more recommended way to deal with this cleanly.

File: Sources/XMLParser/Conversions.swift (hunk 2 above)
Comment (ratranqu) [reply Sources/XMLParser/Conversions.swift:42]: tbh, I am not sure. I fixed based on the type mismatches which posed up.

That said, there hv been some changes in the ParserBuilders (https://github.com/pointfreeco/swift-parsing/discussions/290), which might hint to the reason.

In particular, see how in the PrefixTest we now have to specifically add the Output as second parameter (as a Tuple) to the ParserPrinter, whereas before, it was probably inferred as such:
https://github.com/pointfreeco/swift-parsing/commit/c6e2241daa46e5c6e5027a93b161bca6ba692bcc

Comment (JaapWijnen) [reply Sources/XMLParser/Conversions.swift:42]: Seems like tuples are now unpacked automatically. Which is pretty nice

**Comment by JaapWijnen** [comment]
Sorry forgot to make 1 last mention. Could you update the package versions? Since this is a breaking release I'd also like the swift-parsing package version to be 0.12.0 or whatever the newest is :)

**Comment by JaapWijnen** [comment]
Already merged so that would be a separate PR. I'll cut a release right after that

**Pull request merged**